### PR TITLE
Issue 40 page layoutcleanup

### LINKED ISSUE #39: Creating the General page layout
The page has 3 general distinct segmets, the top header, the sidebar navigation, and the actual dashboard section where the data is displayed.

I think we need to create this as soon as possible.
- [ ] header with the logo and login/logout functions
- [ ] Left sideBar navigation section
- [ ] And a section for the tables and data display

householdsviewsegmented

### LINKED ISSUE #40: Modifying the header and styling to look as in mockups
The header should look like this:
Possible file changes to(NB: not an exhaustive list) :
- the app component
- header component
- and both their styling

This is the goal:
image

- Responsiveness is key

### LINKED ISSUE #41: Modyfing and styling the Navigation to look as in the mockups
part of #39 

Style the navigation to the left sidebar, as shown in the mock up:

image

### LINKED ISSUE #42: styling the dashboard section to look and feel as the mockup design
Part of #39.
Tasks for this __might__ only involve around styling the App component so that we have this section in the page:

image

At the moment the table and the search/filter fields  are not required for this tasks we just need the layout for the 3 main regions as demarcated in issue: #39

### PR DESCRIPTION
**Pull request by p-netm**
Code Refactor and Add tests.

closes #43 , This closes #40, closes #41, closes #42 and also finally closes #39.

### CODE REVIEW

File: src/App/tests/App.test.tsx [hunk 1]
```diff
@@ -36,6 +36,18 @@ describe('App', () => {
...
+
+    // should have a sidebar for the side navigation
+    const sideMenuWrapper = wrapper.find('div.side-menu-section');
+    expect(sideMenuWrapper.length).toEqual(1);
+
+    // should have the main page where the page views are rendered
+    const mainPageWrapper = wrapper.find('div.container-section');
+    expect(mainPageWrapper.length).toEqual(1);
+
     expect(toJson(wrapper)).toMatchSnapshot();
```
Comment (FrankApiyo) [inline src/App/tests/App.test.tsx]: looks good to me except for this snapshot. It contains redundant information about the HeaderComponent you just tested for.

**Comment by p-netm** [comment]
opensrp-web-page-layout

File: src/components/page/SideMenu/SubMenu/index.tsx [hunk 2]
```diff
@@ -0,0 +1,152 @@
+import { IconProp } from '@fortawesome/fontawesome-svg-core';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import classNames from 'classnames';
+import React, { Fragment } from 'react';
+import { RouteComponentProps, withRouter } from 'react-router';
```
Comment (moshthepitt) [inline src/components/page/SideMenu/SubMenu/index.tsx]: `withRouter` is an unused import

**Comment by moshthepitt** [comment]
weird

This looks kind of weird.  What are your thoughts @OluwatobiMayowa ?

**Comment by moshthepitt** [comment]
top

Is it supposed to come to the top of the page like this?

File: src/components/page/SideMenu/SubMenu/index.tsx (hunk 2 above)
Comment (p-netm) [reply src/components/page/SideMenu/SubMenu/index.tsx]: not anymore

**Comment by p-netm** [comment]
> This looks kind of weird. What are your thoughts @OluwatobiMayowa ?

will have to wait for Tobi or Rowo to comment, in the meanwhile ill have login or logout only.

**Comment by p-netm** [comment]
> Is it supposed to come to the top of the page like this?

probably not., adding 24px worth of breathing space

**Comment by moshthepitt** [comment]
> will have to wait for Tobi or Rowo to comment, in the meanwhile, ill leave it like this, unless 

Can you create an issue to address this @p-netm ?

> @moshthepitt you have any suggestions on how to handle this

I think the contents of the dropdown menu need to match the width of the button, and probably the color.  @rowo what do you suggest?

**Comment by moshthepitt** [comment]
> probably not., adding 24px worth of breathing space

@p-netm  Is 24px randomly chose?  Why not go with the standup bootstrap multiples of 15px, (15, 20, 45, etc)?  @rowo @OluwatobiMayowa direction on this would be really appreciated.

**Comment by rowo** [comment]
@moshthepitt @p-netm I was looking at the "dropdown link" example on https://getbootstrap.com/docs/4.0/components/navbar/, but then realized the Bootstrap menu itself tells us what to do. Main problem looks to be the extra space to the right of the username. That username label and carrot need to be right aligned whereas the dropdown box should be wide enough to fall below the icon and its content left aligned.

Screen Shot 2019-10-07 at 3 26 38 PM

**Comment by moshthepitt** [comment]
@p-netm lets try and implement Roger's feedback

**Comment by p-netm** [comment]
> @p-netm Is 24px randomly chosen? Why not go with the standup bootstrap multiples of 15px, (15, 30, 45, etc)? @rowo @OluwatobiMayowa direction on this would be really appreciated.

Got this from the mockup, but something i had to derive.

**Comment by p-netm** [comment]
Further changes including those requested will be done on #40 .

**Pull request merged**